pubtools/pulplib/_impl/model/maintenance.py

```diff
@@ -0,0 +1,200 @@
+import logging
+import datetime
+import os
+
+import jsonschema
+
+from pubtools.pulplib._impl import compat_attr as attr
+from ..schema import load_schema
+from .common import InvalidDataException
+
+
+LOG = logging.getLogger("pubtools.pulplib")
+
+
+USER = os.environ.get("USER")
+HOSTNAME = os.environ.get("HOSTNAME")
+
+
+def iso_time_now():
+    return datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+@attr.s(kw_only=True, frozen=True)
+class MaintenanceEntry(object):
+    """Details about the maintenance status of a specific repository."""
+
+    repo_id = attr.ib(type=str)
+    """ID of repository in maintenance.
+    Note: there is no guarantee that a repository of this ID currenly exists
+          in the Pulp server."""
```

Comment:
Have you published the docs locally and reviewed them?  This doesn't render correctly at the moment due to current whitespace.  Can you please fix that?